### PR TITLE
feature: Add Amazon Inspector support

### PR DESCRIPTION
**Pull request by wvanheerde**
Adds optional Amazon Inspector support with security hub integration delegated to the audit account.

### CODE REVIEW

**Comment by stefanwb** [comment]
On a vanilla deployment this throws an error on `module.landing_zone.aws_inspector2_organization_configuration.default[0]`:

```
Error: updating Amazon Inspector Organization Configuration (000000000000): operation error Inspector2: UpdateOrganizationConfiguration, https response error StatusCode: 400, RequestID: 39bc6fc4-82ef-4a21-ad23-bbd42252186b, ValidationException: Invoking account is not enabled
```

This is caused by Trusted access not being enabled automatically.

Currently looking how to enable this via code on the right account.

**Comment by stefanwb** [comment]
Enabling must happen in a specific order as it has been described here: https://github.com/aws-samples/inspector2-enablement-with-cli

**Comment by stefanwb** [comment]
Put my fixes in a new commit after checking with @wvanheerde.
It LGTM now but @marwinbaumannsbp could you also do a review, just to be sure?

File: examples/basic/.terraform.lock.hcl
Comment (marwinbaumannsbp) [inline examples/basic/.terraform.lock.hcl:1]: Can this file be removed?

Comment (wvanheerde) [reply examples/basic/.terraform.lock.hcl:1]: Done

**Review by marwinbaumannsbp — APPROVED** [review]

**Pull request merged**